### PR TITLE
Allow GC Models to stand in for hashes via Mixin

### PR DESCRIPTION
**Pull request by ajjahn**
My apps typically have bits of state that began as a hash, but evolve into a model. Requiring and including this module makes the transition easier.

### CODE REVIEW

File: lib/grand_central/model/subscriptable.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+require "grand_central/model"
+
+module GrandCentral
+  class Model
+    module Subscriptable
+      def self.included(base)
+        base.send(:alias_method, :merge, :update)
+      end
```
Comment (jgaskins) [inline lib/grand_central/model/subscriptable.rb]: Is there a reason you didn't do this instead?

```ruby
def merge(other)
  update(other)
end
```

**Comment by jgaskins** [comment]
I like the idea of making this a mixin so you have to opt into it explicitly. The `[]` method is genuinely useful on some things, like an object that represents a collection you load from the server:

```ruby
class Collection < GrandCentral::Model
  include Enumerable

  attributes(:models, :loaded, :loading)
  alias loaded? loaded
  alias loading? loading

  def [] id
    models_by_id[id]
  end

  def models_by_id
    # Yes, I'm memoizing inside an otherwise immutable object. :-)
    @models_by_id ||= models.each_with_object({}) { |m, h| h[m.id] = m }
  end
end
```

File: lib/grand_central/model/subscriptable.rb (hunk 1 above)
Comment (ajjahn) [inline lib/grand_central/model/subscriptable.rb]: Hmm, I think it had to do with some swizzling I was doing in my app, but can't remember now. It should be just defined like a regular method. I'll update that.

**Comment by jgaskins** [comment]
I think this is a pretty cool idea, kind of upgrading from a hash to a real object. And you can choose to remove this mixin if you want after you finish converting it, even.

**Pull request merged**